### ISSUE TITLE
release: v19.2.0-beta.20190930

### ISSUE THREAD

**Issue by mjibson** [cockroachdb/cockroach#41128]
Candidate SHA: 250f4c36de2b88eff443cf9be9cd5d2759312c88
Deployment status: http://mjibson-release-250f4c36de2b88eff443cf9be9cd5d2759312c88-0001.roachprod.crdb.io:26258/#/metrics/overview/cluster
Older: http://mjibson-release-v1920-beta20190930-0001.roachprod.crdb.io:26258/
Even older: http://mjibson-release-v1920-alpha20190805-0001.roachprod.crdb.io:26258/

Release qualification: 
- https://teamcity.cockroachdb.com/viewLog.html?buildId=1518053&
- https://teamcity.cockroachdb.com/viewLog.html?buildId=1518030&buildTypeId=Cockroach_ReleaseQualification

Nightly Suite: https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite
Old nightly suite: https://teamcity.cockroachdb.com/viewLog.html?buildId=1511951&buildTypeId=Cockroach_Nightlies_NightlySuite
Even older nightly suite: https://teamcity.cockroachdb.com/viewLog.html?buildId=1509990&buildTypeId=Cockroach_Nightlies_NightlySuite
 
[Release process checklist](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process)
 
Prep date: `2019-09-30`
- [x] [Pick a SHA](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-PickingaSHA), fill in `Candidate SHA` above, notify #release-process of SHA.
- [x] [Tag the provisional SHA](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-TagtheprovisionalSHA)
- [x] [Publish provisional binaries](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-Publishtheprovisionalbinaries)
- [x] [Check binaries](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-Checkbinaries) 
- [x] [Deploy to test clusters](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-DeploytoTestClusters)
- [x] [Verify node crash reports](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-Verifynodecrashreportsappearinsentry.io)
- [ ] [Start release qualification suite](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-RuntheSuiteofNightlyRoachtests)
- [x] [Start nightly suite](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-RuntheSuiteofNightlyRoachtests)
- [x] fill in `Deployment status` above with clusters and `Nightly Suite` with the link to Nightly TeamCity Job
 
- Keep an eye on clusters until release date. Do not proceed below until the release date.
 
Release date: `2019-10-02`
- [x] [Check cluster status](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-Checkclusterstatus)
- [x] [Tag release](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-Tagtherelease)
- [x] [Bless provisional binaries](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-Blesstheprovisionalbinaries)
 
- For production or stable releases in the latest major release series
  - [ ] Update the [Homebrew Formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/cockroach.rb)
  - [ ] Update the image tag in our [Orchestrator configurations](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-Orchestratorconfigurations)
  - [ ] Announce version to the [registration cluster](https://github.com/cockroachlabs/production#publishing-a-new-release-version)
  - [ ] Upgrade the registration cluster (reg runs latest stable version)
 
- [ ] Update docs
- [ ] External communications for release

**Comment by mjibson** [comment]
Changing the SHA to cf5c2bd2372e633d2f63e08e5bffca7c2a7ec59f from 77f26d185efb436aaac88243de19a27caa5da9b6 to pick up the fix for #41145.

**Comment by mjibson** [comment]
Test clusters are still running 77f26d1.

**Comment by mjibson** [comment]
Failed roachtest nightlies:

- [ ] import/tpcc/warehouses=1000/nodes=32
- [ ] restore2TB/nodes=10
- [ ] schemachange/mixed/tpcc
- [ ] tpcc/mixed-headroom/n5cpu16
- [ ] version/mixed/nodes=5
- [ ] cancel/tpcc/distsql/w=10,nodes=3
- [ ] clearrange/checks=false
- [ ] clearrange/checks=true
- [ ] hibernate
- [ ] kv50/rangelookups/relocate/nodes=8
- [ ] tpccbench/nodes=9/cpu=4/chaos/partition
- [ ] typeorm
- [ ] acceptance/bank/zerosum-splits

**Comment by mjibson** [comment]
This release is canceled due to some new bugs. I'm going to start a new roachprod cluster comparison with master as of right now. On Monday the release manager and others will decide what to do. I'll post links to those test clusters here.

**Comment by mjibson** [comment]
New clusters at:
http://mjibson-release-v1920-alpha20190805-0001.roachprod.crdb.io:26258
http://mjibson-release-250f4c36de2b88eff443cf9be9cd5d2759312c88-0001.roachprod.crdb.io:26258

**Comment by mjibson** [comment]
Latency on the master cluster is going up. It is below 20ms for the previous release. Read ops are similar.

![ss](https://user-images.githubusercontent.com/41181/65898630-3addcc00-e36f-11e9-80cd-f582236720cd.png)

**Comment by mjibson** [comment]
This is true for most metrics.

**Comment by mjibson** [comment]
![ss](https://user-images.githubusercontent.com/41181/65898948-e0913b00-e36f-11e9-8c30-3d688a2e68e0.png)

For comparison, the same image of the 0805 release cluster.

**Comment by mjibson** [comment]
Starting a nightly suite for 2c0d5e6241c75cc60a94cc7124dc782b73989893 at https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite

**Comment by mjibson** [comment]
Current status: we are going to try to release 250f4c36de2b88eff443cf9be9cd5d2759312c88 on Wednesday (Oct 2).

**Comment by knz** [comment]
(sorry wrong button)
Regarding the 3rd issue on the list: @lucy-zhang found that VALIDATE CONSTRAINT running concurrently with TPC-C reveals invalid FK relations. The output is OK when TPC-C is not running concurrently. This means that there are isolation problems. 

I am tempted to interpret this as a real-world instance of #41173, @andreimatei what do you think?

**Comment by knz** [comment]
from @lucy-zhang offline:
>  I'm not sure if andrei's issue is the same thing. the rows that are supposed to be missing are in tpcc.warehouse, which we never update (AFAIK) after we restore the fixtures, so even if the read timestamp were slightly behind for some parts of the reads it shouldn't matter

**Comment by knz** [comment]
Failed nightlies:

- [ ] [cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId-9001178873075005748) 
- [ ] [clearrange/checks=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId4040650493650131270)
- [ ] [import/tpcc/warehouses=1000/nodes=32](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId-4551784729715115627)
- [ ] [network/tpcc/nodes=4](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId285598859115364203)
- [ ] [restore2TB/nodes=10](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId741854490244253129)
- [ ] [clearrange/checks=false](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId-7549905676748780804)
- [ ] [hibernate](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId-3931757827174594180)
- [ ] [kv50/rangelookups/relocate/nodes=8](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId7808114564804200759)
- [ ] [schemachange/mixed/tpcc](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId5401367090847010053)
- [ ] [TestRandomSyntaxSQLSmith](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514704&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId-1657705641918481982)

**Comment by knz** [comment]
For reference, the following failures from the original list on 2019-09-27 are not there any more:

-  tpcc/mixed-headroom/n5cpu16
-  version/mixed/nodes=5
-  tpccbench/nodes=9/cpu=4/chaos/partition
-  typeorm
-  acceptance/bank/zerosum-splits

The following are new:

- network/tpcc/nodes=4
- TestRandomSyntaxSQLSmith (although I suspect we're going to ignore this since this test is dripping issues by design)

**Comment by knz** [comment]
My analysis of the issues (same order as above):

- #38417: cancel test fails with `query completed before it could be canceled`. I have not look at the test source but it may be a bug in the test instead of a bug in cockroachdb. 
  - My assessment: **not release blocker** +  further investigation needed.
- #38720  + #41123: import times out before the test can start.
  - My assessment: **unsure**. The error means that the `clearrange` test does not even get to run. Do we have confidence that the code that the test is checking is OK otherwise? @dt please provide input. If the `clearrange` test has not ran for a long time, that would be a release blocker.
  - Also, we need to change the TC reporting so that a test that does not even start does not get marked as "failed" (and instead keep metric of how often import is holding up a test). @kenliu what would be a good way to put this on the dev-inf radar? Should I file a separate issue?
- #40538: hibernate failure. This time we see `373 tests failed unexpectedly` instead of the usual `NN tests succeeded unexpectedly`. This is likely to mark a regression.
  - My assessment: **maybe release blocker** (depending on the failures). @rafiss @jordanlewis please provide input.
  - Also, we need to change the way this roachtest reports its failures, so that different issues get created for "unexpected success" and "unexpected failure".
- #41154: import times out before the test can start. Same as #38720 / #41123 above. However this time the test is about `import`, not about `clearrange`.
  - My assessment: **not release blocker** - we know that import succeeds often enough because many other tests use tpcc.
- #40359: something about the output of the kv workload. **I don't understand the failure log**. @nvanbenschoten can you explain what's going on?
  - My assessment: **unsure** + more investigation needed.
- #37259: inter-node network mishaps cause a tpcc run to fail.
  - My understanding: the test uses toxiproxy between nodes to introduce network partitions, and then asserts that there is no goroutine peak in the server. However, the partition also causes legitimate SQL errors in clients, and that makes the test abort in error, in a way that's not relevant to the primary thing being tested. @ajwerner do you agree with my analysis?
  - My assessment: **not release blocker** if my understanding is correct
  - Also if the analysis is correct the test needs to be fixed.
- #41152: restore 2TB fails. This seems to be an import timeout, like #41154 and #38720 / #41123.
  - My assessment: same as for the clearrange above: **unsure**, this is not a release blocker unless we have confidence that the primary feature exercised by the test has been tested to work OK often enough. @dt please provide input.
- #40935: a VALIDATE CONSTRAINTS concurrent with large tpc-c finds erroneous rows. There's a repro readily available.
  - My assessment: **probably release blocker** if this confirms to be a serializability violation (isolation error)
  - Discussed with @lucy-zhang: need to bisect to see where the problem was introduced. Determine relationship with other issue #41173 found by Andrei, perhaps by running the tpc-c load with `distsql=off`.

**Comment by knz** [comment]
Regarding this:

> #37259: inter-node network mishaps cause a tpcc run to fail.
> My understanding: the test uses toxiproxy between nodes to introduce network partitions, and then asserts that there is no goroutine peak in the server. However, the partition also causes legitimate SQL errors in clients, and that makes the test abort in error, in a way that's not relevant to the primary thing being tested. @ajwerner do you agree with my analysis?

Answer from @ajwerner:

> That initial analysis seems sound.

**Comment by nvanbenschoten** [comment]
#40359 can be ignored. It appears to be a testing error caused by new retryable errors that we see during RELOCATE RANGE statements. See https://github.com/cockroachdb/cockroach/pull/41106 (which I'm planning to return to later today).

**Comment by jordanlewis** [comment]
Signing off on the cancel test. It's a testing error caused by the operation in question finishing too quickly.

**Comment by knz** [comment]
Regarding #40935  update from @lucy-zhang 

> current update is that @jordan and I have tried to reproduce this by turning loadgen back on for one of the clusters where I got this test failure, and running that select query manually, and neither of us have seen it again
> so it looks like there is some additional state necessary to repro aside from "heavy load" (possibly related to the early state of the cluster?)
> so this does seem rarer than I first expected

Separately @andy-kimball states with agreement from @bdarnell 

> If this repros only under heavy stress, and we have seen it nowhere else, I don't think it should be classified as a beta blocker. The bar should be very high now. I'd only block beta for "high severity" (which this is) and "common" (which this isn't).

So I'm checking this as signed off by lucy, jordan and andy.

**Comment by rafiss** [comment]
The Hibernate issues (#40538) are not a beta blocker. The tests are concerningly flaky though, and we will continue investigating this during the rest of the release period. The issues stem from the Hibernate tests occasionally being unable to connect to the DB.

**Comment by mjibson** [comment]
I checked off the sqlsmith failure because it should never block a release.

**Comment by knz** [comment]
There's discussion on to whether to adopt the latest changes that improve on #41206 (perf regression).
If we bump the SHA here is the diff:
#40493
#40948
#40954
#41129
#41138
#41153
#41187
#41190
#41192
#41194
#41195
#41196
#41212
#41215
#41217
#41220
#41221
#41231
#41235
#41237
#41241
#41244

**Comment by knz** [comment]
Input from @awoods187 and @bdarnell: continue with the same SHA.

**Comment by knz** [comment]
Note I have checked the cluster health at http://mjibson-release-250f4c36de2b88eff443cf9be9cd5d2759312c88-0001.roachprod.crdb.io:26258/#/metrics/overview/cluster

The cluster displays a performance anomaly for the period up to and including Sept 30.

Then yesterday (Oct 1) Nathan uploaded a new binary with the fix from #41220 (and other fixes) which demonstrates that the anomaly disappears. **Note however that these fixes are not present in today's release.**

**Comment by knz** [comment]
@dt about the `clearrange` tests:
> AFAIK, it is just slow, not wrong.

**Comment by knz** [comment]
@dt about the `restore2TB` test:
> the RESTORE one I took a very cursory look at and didn’t see what killed it, just exit 255
> might have OOMed I guess — i wonder if it was the rocks logging leak? I donno. I’m fine signing off on that too I guess.

**Comment by knz** [comment]
@irfansharif and @andreimatei do not have anything to say on the remaining roachtest failure. 

So I did go and investigate the log files. I am not seeing errors in the CockroachDB logs themselves other than the expected "cannot connect to node" (the node being shut down under chaos).

@bdarnell says "go"

**Comment by knz** [comment]
The version is tagged, the binaries are uploaded, the docker image works, so from engineering the release is ready to go.
The release notes PR is here: https://github.com/cockroachdb/docs/pull/5250

Handing this off to docs.
